fetch-in-react-practice/client/components/app.jsx

```diff
@@ -81,12 +81,22 @@ export default class App extends React.Component {
       .then(res => {
         return res.json()
           .then(data => {
-            if (res.ok) return obj;
+            if (res.ok) {
+              return obj;
+            }
             return { ...obj, error: data };
-          });
-      })
-      .catch(err => console.error(err));
-    this.componentDidMount();
+          })
+          .then(result => {
+            const newArr = [...this.state.todos];
+            for (let i = 0; i < newArr.length; i++) {
+              if (newArr[i].todoId === todoId) {
+                newArr[i].isCompleted = !newArr[i].isCompleted;
```

Comment:
instead of slightly tweaking the object in the array, just replace it completely